packages/cli/cli/src/commands/upgrade-generator/upgradeGeneratorInWorkspace.ts

```diff
@@ -0,0 +1,22 @@
+import { upgradeGeneratorsIfPresent } from "@fern-api/manage-generator";
+import { loadRawWorkspaceConfiguration } from "@fern-api/workspace-configuration";
+import { writeFile } from "fs/promises";
+import yaml from "js-yaml";
+import { loadProject } from "../utils/load-project/loadProject";
+
+export async function upgradeGeneratorsInWorkspace(commandLineWorkspaces: readonly string[]): Promise<void> {
+    const { workspaceConfigurations } = await loadProject({ commandLineWorkspaces });
+
+    for (const workspaceConfigurationFilePath of workspaceConfigurations) {
+        const workspaceConfiguration = await loadRawWorkspaceConfiguration(workspaceConfigurationFilePath);
+        const updatedWorkspaceConfiguration = upgradeGeneratorsIfPresent({ workspaceConfiguration });
+        if (updatedWorkspaceConfiguration.updatedWorkspaceConfiguration !== workspaceConfiguration) {
```

Comment:
This is testing reference equality, you'd need to do a deep equals. But I think we can just skip this check

Reply:
ah makes sense, i just stole it from addGenerator

Reply:
Ah cool, we should fix that too at some point